### PR TITLE
Adds more tracking to the reader, comments, and notifications

### PR DESCRIPTION
**Pull request by emilylaguna**
Project: #17503

### Description
Adds new tracking to the Reader, Comments, and Notifications:

#### Reader
- When the user clears their search history
- Tracks the source and type when performing a search
- Removed a duplicated event when toggling the follow notifications
- When tapping on a link and image on the reader detail

#### Comments
- When entering or exiting fullscreen in the comment reply field

#### Notifications
- When tapping the up and down arrows on the notifications view

#### Reader Search
1. Launch the app
2. Tap the Reader tab
3. Tap the Search icon in the top corner of the view
4. Perform a search, and verify you see: `🔵 Tracked: reader_search_performed <source: user_input, type: posts>`
5. Tap on the Sites tab
6. Perform a search, and verify you see: `🔵 Tracked: reader_search_performed <source: user_input, type: sites>`
7. Clear the search text
8. Tap on an item from your search history, and verify you see: `🔵 Tracked: reader_search_performed <source: search_history, type: posts>`
9. Tap on the search field again
10. Tap the 'Clear search history' button, and verify you see: `🔵 Tracked: reader_search_history_cleared <subscription_count: COUNT>`

#### Reader Toggle Follow
1. Launch the app
2. Tap the Reader tab
3. Locate an item in the feed
4. Tap the more ⠇ button
5. Tap the 'Turn on site notifications' button
6. Verify you see only 1 `🔵 Tracked: followed_blog_notifications_reader_menu_on <blogId: BLOG_ID, subscription_count: COUNT>`
7. Toggle the option off and verify you see only 1 `🔵 Tracked: followed_blog_notifications_reader_menu_off <blogId: BLOG_ID, subscription_count: COUNT>`

#### Comments
1. Launch the app
2. Tap the Reader tab
3. Tap the comment button to compose a comment
4. Tap the up arrow, and verify you see: `🔵 Tracked: comment_fullscreen_entered <>`
5. Tap the down arrow and verify you see `🔵 Tracked: comment_fullscreen_exited <>`

#### Notifications
1. Launch the app
2. Tap the Notifications tab
3. Tap on an item to view the details
4. Tap the down arrow and verify you see `🔵 Tracked: notifications_previous_tapped <>`
5. Tap the up arrow and verify you see `🔵 Tracked: notifications_next_tapped <>`

To me it feels a little weird that the down arrow is previous and up is next, this matches the naming in code. 

### Regression Notes
1. Potential unintended areas of impact
None, adding tracking.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
None.

### PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Reader/ReaderSearchViewController.swift [hunk 1]
```diff
@@ -20,6 +20,25 @@ import Gridicons
...
+        case userInput
+        case searchHistory
+
+        var value: String {
+            switch self {
+            case .userInput: return "user_input"
+            case .searchHistory: return "search_history"
+            }
+        }
     }
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Reader/ReaderSearchViewController.swift]: Could this be simplified a little to:

```
private enum SearchSource: String {
    case userInput = "user_input"
    case searchHistory = "search_history"
}
```

and then use `rawValue` where we were using `value`?

**Review by frosty — COMMENTED** [review]
> Tap the 'Turn on site notifications' button

It took me a while to find a post that had this option, and I'm not sure why!

PR looks good! Just one small code comment!

**Comment by mokagio** [comment]
@emilylaguna  I'm going to bump this to the next release because we'll be code freezing 18.8 today and this hasn't been approved yet. @frosty basically said this is good to go, but there are also conflicts against `develop`.

If this cannot wait two weeks and it's important that it makes it into this release, let me know and we'll organize a new beta once ready.

File: WordPress/Classes/ViewRelated/Reader/ReaderSearchViewController.swift (hunk 1 above)
Comment (emilylaguna) [reply WordPress/Classes/ViewRelated/Reader/ReaderSearchViewController.swift]: That's a way better idea than what I had. Updated the code: 035d8dc2415357528a7aa89a222a0b575e5f7c77

**Review by frosty — APPROVED** [review]
:shipit:

**Pull request merged**